pkg/controller/core/queuedworkload_controller.go

```diff
@@ -58,7 +63,40 @@ func NewQueuedWorkloadReconciler(queues *queue.Manager, cache *cache.Cache) *Que
 //+kubebuilder:rbac:groups=kueue.x-k8s.io,resources=queuedworkloads/finalizers,verbs=update
 
 func (r *QueuedWorkloadReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
-	// No-op. All work is done in the event handler.
+	var wlObj kueue.QueuedWorkload
+	if err := r.client.Get(ctx, req.NamespacedName, &wlObj); err != nil {
+		// we'll ignore not-found errors, since there is nothing to do.
+		return ctrl.Result{}, client.IgnoreNotFound(err)
+	}
+	log := ctrl.LoggerFrom(ctx).WithValues("queuedWorkload", klog.KObj(&wlObj))
+	ctx = ctrl.LoggerInto(ctx, log)
+	log.V(2).Info("Reconciling QueuedWorkload")
+
+	status := workloadStatus(&wlObj)
+	var err error
+	switch status {
+	case pending:
+		if !r.queues.QueueExists(&wlObj) {
+			err = r.UpdateWorkloadStatus(ctx, &wlObj, kueue.QueuedWorkloadAdmitted, corev1.ConditionFalse,
+				fmt.Sprintf("Queue: %s for workload didn't exist", wlObj.Spec.QueueName),
+				"QueueMissing")
+			return ctrl.Result{}, err
+		}
+		err = r.UpdateWorkloadStatus(ctx, &wlObj, kueue.QueuedWorkloadAdmitted, corev1.ConditionFalse,
+			"Waiting to be admitted",
+			"Queued")
+		return ctrl.Result{}, err
```

Comment:
What if the scheduler decided that it doesn't fit? Sure, it's still queued, but we would like to say that we tried to schedule it and it didn't fit.

Reply:
that should go on the scheduler side, same as job controller is adding the Finished bit, the scheduler bit could help here